param/coercer/coercer.go

```diff
@@ -19,57 +19,35 @@ func CoerceParams(schema *spec.Schema, data map[string]interface{}) error {
 		if !ok {
 			continue
 		}
-
-		valMap, ok := val.(map[string]interface{})
-		if ok {
-			CoerceParams(subSchema, valMap)
-
-			if subSchema.Type == arrayType {
-				valSlice, err := parseIntegerIndexedMap(valMap)
```

Comment:
Converting integer-indexed map to array is handle within `coerceNonObjectSchema` together with other array-related logic